source/Stargate-REST-API/RESTfulWebService.class.st

```diff
@@ -0,0 +1,108 @@
+"
+I represent a RESTful Web Service, and provide the base support to easily create specific services.
+
+I provide a way to encode/decode the resources into the representations produced/consumed by the service.
+I also provide standard error handling and templates for processing the differente HTTP verbs.
+"
+Class {
+	#name : #RESTfulWebService,
+	#superclass : #Object,
+	#instVars : [
+		'mappingRuleSet'
+	],
+	#category : #'Stargate-REST-API-Model'
+}
+
+{ #category : #'private - transformations' }
+RESTfulWebService >> decode: theRepresentation at: aKey from: sourceMediaType within: aContext [
+
+	^ [ [ (mappingRuleSet ruleToDecode: sourceMediaType to: aKey) applyOn: theRepresentation within: aContext ]
+		on: KeyNotFound , DecodingFailed
+		do: [ :signal | HTTPClientError signalBadRequest: ('Expected key <1s> was not found' expandMacrosWith: signal key) ] ]
+		on: MappingNotFound
+		do: [ :signal | HTTPClientError signalUnsupportedMediaType: signal messageText ]
+]
+
+{ #category : #'private - transformations' }
+RESTfulWebService >> encode: theResource at: aKey to: targetMediaType within: aContext [
```

Comment:
The at: part of the message bothers me, it makes no sense.

encode: theResource to: targetMediaType usingMappingAt: aKey within: aContext

decode: theRepresentation from: sourceMediaType usingMappingAt: aKey within: aContext